src/main/rx/nrf24_syma.c

```diff
@@ -0,0 +1,293 @@
+
+    NRF24L01_WriteReg(NRF24L01_01_EN_AA, 0); // No auto acknowledgment
+    NRF24L01_WriteReg(NRF24L01_02_EN_RXADDR, BV(NRF24L01_02_EN_RXADDR_ERX_P0));
+    NRF24L01_WriteReg(NRF24L01_03_SETUP_AW, NRF24L01_03_SETUP_AW_5BYTES);   // 5-byte RX/TX address
+    if (symaProtocol == NRF24RX_SYMA_X) {
+        payloadSize = SYMA_X_PROTOCOL_PAYLOAD_SIZE;
+        NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, NRF24L01_06_RF_SETUP_RF_DR_250Kbps | NRF24L01_06_RF_SETUP_RF_PWR_n12dbm);
+        protocolState = STATE_BIND;
+        // RX_ADDR for pipes P1-P5 are left at default values
+        NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rxTxAddr, RX_TX_ADDR_LEN);
+    } else {
+        payloadSize = SYMA_X5C_PROTOCOL_PAYLOAD_SIZE;
+        NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, NRF24L01_06_RF_SETUP_RF_DR_1Mbps | NRF24L01_06_RF_SETUP_RF_PWR_n12dbm);
+        // RX_ADDR for pipes P1-P5 are left at default values
+        NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rxTxAddrX5C, RX_TX_ADDR_LEN);
+        // just go straight into data mode, since the SYMA_X5C protocol does not actually require binding
+        protocolState = STATE_DATA;
+        rfChannelCount = SYMA_X5C_RF_CHANNEL_COUNT;
+        memcpy(rfChannels, rfChannelsX5C, SYMA_X5C_RF_CHANNEL_COUNT);
+    }
+    NRF24L01_SetChannel(rfChannels[0]);
+
+    NRF24L01_WriteReg(NRF24L01_08_OBSERVE_TX, 0x00);
+    NRF24L01_WriteReg(NRF24L01_1C_DYNPD, 0x00); // Disable dynamic payload length on all pipes
+
+    NRF24L01_WriteReg(NRF24L01_11_RX_PW_P0, payloadSize);
+
+    NRF24L01_SetRxMode(); // enter receive mode to start listening for packets
+}
+
+STATIC_UNIT_TESTED uint16_t convertToPwmUnsigned(uint8_t val)
+{
+    uint32_t ret = val;
+    ret = ret * (PWM_RANGE_MAX - PWM_RANGE_MIN) / UINT8_MAX + PWM_RANGE_MIN;
+    return (uint16_t)ret;
+}
+
+STATIC_UNIT_TESTED uint16_t convertToPwmSigned(uint8_t val)
+{
+    int32_t ret = val & 0x7f;
+    ret = (ret * (PWM_RANGE_MAX - PWM_RANGE_MIN)) / (2 * INT8_MAX);
+    if (val & 0x80) {// sign bit set
+        ret = -ret;
+    }
+    return (uint16_t)(PWM_RANGE_MIDDLE + ret);
+}
+
+STATIC_UNIT_TESTED bool checkBindPacket(const uint8_t *packet)
+{
+    bool bindPacket = false;
+    if (symaProtocol == NRF24RX_SYMA_X) {
+        if ((packet[5] == 0xaa) && (packet[6] == 0xaa) && (packet[7] == 0xaa)) {
+            bindPacket = true;
+            rxTxAddr[4] = packet[0];
+            rxTxAddr[3] = packet[1];
+            rxTxAddr[2] = packet[2];
+            rxTxAddr[1] = packet[3];
+            rxTxAddr[0] = packet[4];
+        }
+    } else {
+        if ((packet[0] == 0) && (packet[1] == 0) && (packet[14] == 0xc0) && (packet[15] == 0x17)) {
+            bindPacket = true;
+        }
+    }
+    return bindPacket;
+}
+
+void symaSetRcDataFromPayload(uint16_t *rcData, const uint8_t *packet)
+{
+    for (int ii = NRF24_AUX1; ii <= NRF24_AUX5; ++ii) {
+        rcData[ii] =  PWM_RANGE_MIN;
```

Comment:
Using ternary operators will allow writing `rcData` only once, it is slightly more robust/elegant. And code readability will be about the same
```
rcData[NRF24_AUX5] = packet[4] & 0x80 ? PWM_RANGE_MAX : PWM_RANGE_MIN;
```